src/Algebra/Ordered/Structures.agda

```diff
@@ -166,6 +166,43 @@ record IsProsemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂)
 
   open IsSemiring isSemiring public using (distribˡ; distribʳ; zeroˡ; zeroʳ)
 
+-- Preordered IdempotentSemiring (IdempotentProsemiring)
+
+record IsIdempotentProsemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
+  field
+    isProsemiring  : IsProsemiring + * 0# 1#
+    +-idem         : Idempotent +
+
+  open IsProsemiring isProsemiring public
+
+  isIdempotentSemiring : IsIdempotentSemiring + * 0# 1#
+  isIdempotentSemiring = record { isSemiring = isSemiring ; +-idem = +-idem }
+
+  open IsIdempotentSemiring isIdempotentSemiring public using (+-idem)
+
+-- Preordered KleeneAlgebra (prokleenealgebra)
+
+record IsProkleenealgebra (+ * : Op₂ A) ( ⁻* : Op₁ A) (0# 1# : A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
+  field
+    isIdempotentProsemiring : IsIdempotentProsemiring + * 0# 1#
+    starLeftExpansion       : StarLeftExpansion 1# + * ⁻*
+    starRightExpansion      : StarRightExpansion 1# + * ⁻*
```

Comment:
We usually bundle left and right versions into a single pair, e.g. `StarExpansion` and `FixedPoint`